data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandlerTest.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.javalin.http.Context;
+import java.util.List;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconBlockResponse;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.datastructures.util.DataStructureUtil;
+import tech.pegasys.artemis.provider.JsonProvider;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.HistoricalChainData;
+import tech.pegasys.artemis.storage.Store;
+import tech.pegasys.artemis.util.async.SafeFuture;
+
+public class BeaconBlockHandlerTest {
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private final ChainStorageClient storageClient = mock(ChainStorageClient.class);
+  private final HistoricalChainData historicalChainData = mock(HistoricalChainData.class);
+  private final Store store = mock(Store.class);
+  private final Bytes32 blockRoot = Bytes32.random();
+  private final Context context = mock(Context.class);
+
+  private final SignedBeaconBlock signedBeaconBlock =
+      DataStructureUtil.randomSignedBeaconBlock(1, 1);
+  private final BeaconBlock beaconBlock = signedBeaconBlock.getMessage();
+  private final BeaconBlockHandler handler =
+      new BeaconBlockHandler(storageClient, historicalChainData, jsonProvider);
+
+  @Test
+  public void shouldReturnNotFoundWhenRootQueryAndStoreNull() throws Exception {
+    final String rootKey = "0xf22e4ec2";
+    final Map<String, List<String>> params = Map.of("root", List.of(rootKey));
+
+    when(storageClient.getStore()).thenReturn(null);
+    when(context.queryParamMap()).thenReturn(params);
+
+    handler.handle(context);
+
+    verify(context).status(SC_NOT_FOUND);
+  }
+
+  @Test
+  public void shouldReturnNotFoundWhenValidParamNotSpecified() throws Exception {
```

Comment:
shouldReturnNotFoundWhenParametersNotSpecified

This is a similar use case to #1218, at least yours is returning a 404, not 500. could potentially return a 400 in this use case...